### PR TITLE
FLAG_ACTIVITY_NEW_TASK re-added to floating windows

### PR DESCRIPTION
**Pull request by corcoran**
ActivityStack modified to not cancel results if window is floating and "new".
## 

Aaron merged the original patch for testing -- however he also mentioned that removing FLAG_ACTIVITY_NEW_TASK causes some apps (G+?) to not quit halo when pressing outside of the window.  Should that be the case with the changes I made that were recently merged, _this_ should return the behavior back to what was originally intended as well as address the activity result cancellation issue the original patch was created to fix.

### CODE REVIEW

**Comment by kaluoshi** [comment]
So this is better @corcoran ? lol. I mean should fix the issue without breaking anything

**Comment by corcoran** [comment]
Well, the previous patch worked great for me & I couldn't reproduce the issues Aaron mentioned.  That patch is also cleaner since floating windows behave exactly the same as non-floating windows in that FLAG_ACTIVITY_NEW_TASK isn't given out to _every_ floating intent regardless of if it's actually a new task or not.

That said, and this is speculation, but what I suspected could be happening as per Aaron's response is that some notification intents, when loaded with FLAG_ACTIVITY_NEW_TASK would completely close after pressing outside the floating window, rather than taking you back to say a contact listing/activity feed page.  If that's the case, then this patch works around _that_ issue.

So basically, test what is currently merged & see if notifications launched in a floating window are closing properly when clicking outside of the floating window.  If they're not then this patch will fix that.

Hopefully I didn't make that sound too confusing ;)

**Comment by kaluoshi** [comment]
Maybe a bit confusing lol. Anyway i'll test the merged one later (also with G+ app and other system's and not apps). Will let you know :)
